The test report will not be generated if the folder to save the report doesn't exist

**What happened**:
When the `--report=xml` option is provided, if the folder where the report will be created doesn't exist, `kuttl` fails. This is the error output:
```sh
=== CONT  kuttl
    harness.go:399: run tests finished
    harness.go:508: cleaning up
    harness.go:563: removing temp folder: ""
    harness.go:508: cleaning up
    harness.go:563: removing temp folder: ""
    harness.go:589: fatal error writing report: open tests/_build/artifacts/kuttl-test.xml: no such file or directory
--- FAIL: kuttl (62.16s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/awesometest (1.22s)
```

**What you expected to happen**:
The folder should be created

**How to reproduce it (as minimally and precisely as possible)**:
Run a test, provide the `--report=xml` parameter and don't create the folder where the reports will be generated.

--- a/pkg/report/report.go
+++ b/pkg/report/report.go
@@ -5,6 +5,7 @@ import (
 	"encoding/xml"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"path/filepath"
 	"time"
 )
@@ -216,6 +217,15 @@ func latestEnd(start time.Time, testcases []*Testcase) time.Time {
 // Report prints a report for TestSuites to the directory.  ftype == json | xml
 func (ts *Testsuites) Report(dir, name string, ftype Type) error {
 	ts.Close()
+
+	// Create the folder to save the report if it doesn't exist
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		dirCreationErr := os.MkdirAll(dir, 0755)
+		if dirCreationErr != nil {
+			return dirCreationErr
+		}
+	}
+
 	// if a report is requested it is always created
 	switch ftype {
 	case XML:

--- a/pkg/report/report.go
+++ b/pkg/report/report.go
@@ -219,11 +219,15 @@ func (ts *Testsuites) Report(dir, name string, ftype Type) error {
 	ts.Close()
 
 	// Create the folder to save the report if it doesn't exist
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		dirCreationErr := os.MkdirAll(dir, 0755)
-		if dirCreationErr != nil {
-			return dirCreationErr
+	_, err := os.Stat(dir)
+
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(dir, 0755)
+		if err != nil {
+			return err
 		}
+	} else if err != nil {
+		return err
 	}
 
 	// if a report is requested it is always created

